numba/cuda/api.py

```diff
@@ -231,8 +231,32 @@ def _fill_stride_by_order(shape, dtype, order):
 def device_array_like(ary, stream=0):
     """Call cuda.devicearray() with information from the array.
     """
-    return device_array(shape=ary.shape, dtype=ary.dtype,
-                        strides=ary.strides, stream=stream)
+    # Avoid attempting to recompute strides if the default strides will be
+    # sufficient to create a contiguous array.
+    if ary.is_c_contiguous() or ary.ndim <= 1:
+        return device_array(shape=ary.shape, dtype=ary.dtype, stream=stream)
+
+    # Otherwise, we need to compute new strides using an algorithm adapted from
+    # Numpy's PyArray_NewLikeArrayWithShape in core/src/multiarray/ctors.c. We
```

Comment:
This also could do with an explicit reference with a commit SHA from NumPy.

Also `s/Numpy/NumPy/`

Reply:
I've added a reference to a specific version rather than a SHA - the parts of that function from which this algorithm is derived haven't changed since 2011, so there will also be many versions in which the same algorithm can be found.

Reply:
Sounds good, thanks.